src/map.rs

```diff
@@ -937,6 +937,74 @@ where
         n.leading_zeros() as isize | (1 << (RESIZE_STAMP_BITS - 1)) as isize
     }
 
+    // TODO
+    fn try_presize<'g>(&self, size: usize, guard: &'g Guard) {
+        let new_capacity = if size >= MAXIMUM_CAPACITY / 2 {
+            MAXIMUM_CAPACITY
+        } else {
+            // 1.5 * size + 1
+            let size = size + (size >> 1) + 1;
+
+            std::cmp::min(MAXIMUM_CAPACITY, size.next_power_of_two())
+        };
+
+        let c = new_capacity as isize; // TODO
+
+        loop {
+            let sc = self.size_ctl.load(Ordering::SeqCst);
+            if sc < 0 {
+                break;
+            }
+
+            let table = self.table.load(Ordering::SeqCst, &guard);
+
+            // TODO: clean this up?
+            let n = if table.is_null() {
+                None
+            } else {
+                Some(unsafe { table.deref() }.len())
+            };
+
+            if n.is_none() || n.unwrap() == 0 {
+                let n = sc.max(c) as usize;
+
+                let mut table = self.table.load(Ordering::SeqCst, guard);
+
+                if self.size_ctl.compare_and_swap(sc, -1, Ordering::SeqCst) == sc {
+                    if self.table.load(Ordering::SeqCst, guard) == table {
+                        let new_table = Owned::new(Table::new(n));
+                        table = new_table.into_shared(guard);
+                        self.table.store(table, Ordering::SeqCst);
```

Comment:
_Technically_ this should use `swap` instead, and the old table should be marked as garbage. See [this code in `transfer`](https://github.com/jonhoo/flurry/blob/36ddd9285f8c91b4a82591582f19c61505f84638/src/map.rs#L721). Now, realistically we never (I don't think) allocate a table with 0 bins, so by arriving here (thanks to the `if` above) it is most likely that `self.table` _was_ `null`. We should _either_ `assert!` that the old table value was `null` here, or we should properly mark a non-`null` result from `.swap` as garbage.

Reply:
I think I'll just`assert` for now and see about marking the result as garbage later